api/src/test/java/org/openmrs/module/pihcore/task/PihCloseStaleVisitsTaskTest.java

```diff
@@ -519,7 +519,55 @@ public void test_shouldNotKeepVisitOpenIfItHasEDTriageAndEncounterWithinFortyEig
 
 
     @Test
-    public void test_shouldNotCloseVisitIfDateCreatedWithin24Hours() throws Exception {
+    public void test_shouldCloseVisitIfItHasADischargeEncounterEvenIfDateCreatedAndDateChangedAreWithin12Hours() throws Exception {
+
+        ContextSensitiveMetadataTestUtils.setupDispositionDescriptor(conceptService, dispositionService);
+        ContextSensitiveMetadataTestUtils.setupAdmissionDecisionConcept(conceptService, emrApiProperties);
+        ContextSensitiveMetadataTestUtils.setupSupportsVisitLocationTag(locationService);
+
+        Patient patient = patientService.getPatient(7);    // patient already has one visit in test dataset
+
+        // need to tag the unknown location so we don't run into an error when testing against the existing visits in the test dataset
+        Location unknownLocation = locationService.getLocation(1);
+        unknownLocation.addTag(emrApiProperties.getSupportsVisitsLocationTag());
+        locationService.saveLocation(unknownLocation);
+
+        Location location = locationService.getLocation(2);
+        location.addTag(emrApiProperties.getSupportsVisitsLocationTag());
+        locationService.saveLocation(location);
+
+        Visit visit = new Visit();
+        visit.setDateCreated(new Date());
+        visit.setDateChanged(new Date());
+        visit.setStartDatetime(DateUtils.addHours(new Date(), -60));
+        visit.setPatient(patient);
+        visit.setLocation(location);
+        visit.setVisitType(emrApiProperties.getAtFacilityVisitType());
+
+        // create an exit from care encounter
+        Encounter exitFromCareEncounter = new Encounter();
+        exitFromCareEncounter.setPatient(patient);
+        exitFromCareEncounter.setEncounterType(encounterService.getEncounterTypeByUuid("b6631959-2105-49dd-b154-e1249e0fbcd7"));
```

Comment:
You couldn't get it by name?

Reply:
Even better, turns out I can just get it by EmrApiProperties, I just updated...